test/sql-luatest/datetime_test.lua

```diff
@@ -2387,3 +2387,296 @@ g.test_datetime_32_2 = function()
         t.assert_equals(err.message, res)
     end)
 end
+
+-- Make sure cast from MAP to DATETIME works as intended.
+
+--
+-- The result of CAST() from MAP value to DATETIME must be equal to the result
+-- of calling require('datetime').new() with the corresponding table as an
+-- argument.
+--
+g.test_datetime_33_1 = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local dt = require('datetime')
+        local v = setmetatable({}, { __serialize = 'map' })
+        local sql = [[SELECT CAST(#v AS DATETIME);]]
+        t.assert_equals(box.execute(sql, {{['#v'] = v}}).rows, {{dt.new(v)}})
+
+        v.something = 1
```

Comment:
I would rather throw an error on unknown fields. Or is it the same in Lua?

Reply:
Yes, it is the same for Lua:
```
tarantool> require('datetime').new()
---
- 1970-01-01T00:00:00Z
...

tarantool> require('datetime').new({years = 123})
---
- 1970-01-01T00:00:00Z
...
```